hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -131,6 +131,9 @@ public static String getRecordPartitionPath(GenericRecord record, List<String> p
       } else {
         if (encodePartitionPath) {
           fieldVal = PartitionPathEncodeUtils.escapePathName(fieldVal);
+        } else {
+          // Hive doesn't respect the space at the end, so remove it to avoid duplicate keys error
+          fieldVal = fieldVal.trim();
         }
```

Comment:
Does other engines follow this pattern ?

Reply:
Haven't tested other engines besides HMS

Reply:
I think it's ok to error out if other engines don't support trailing spaces. We can check with Presto/Trino.